### PR TITLE
feat: update gh action to add the logic to open a pull request

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes
This PR adds the logic to open a PR with the generated Spec file. I tested my changes on a fork https://github.com/andreaangiolillo/openapi-test/actions/runs/10064027178

Here is an example of the generated PR: https://github.com/andreaangiolillo/openapi-test/pull/5

### CODE REVIEW

File: .github/workflows/release-spec-prod.yml [hunk 1]
```diff
@@ -19,19 +20,51 @@ jobs:
...
+            labels: failed-release
+            title: "(PROD) Release: `foascli split` command failed :scream_cat:"
+            body: See https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+            token: ${{ secrets.api_bot_pat }}
+
+        - name: Create Pull Request
+          uses: peter-evans/create-pull-request@v6
+          with:
+            token: ${{ secrets.GITHUB_TOKEN }}
+            title: "release: (PROD) Release OpenAPI Spec :rocket:"
```
Comment (jeroenvervaeke) [inline .github/workflows/release-spec-prod.yml:56]: small suggestion: you could add a label to the PR to easily be able to filter out all automated PRs

**Review by jeroenvervaeke — APPROVED** [review]
LGTM, got one small suggestion

File: .github/workflows/release-spec-prod.yml (hunk 1 above)
Comment (andreaangiolillo) [reply .github/workflows/release-spec-prod.yml:56]: good point! I will add it

Comment (andreaangiolillo) [reply .github/workflows/release-spec-prod.yml:56]: updated

**Pull request merged**